src/main/kotlin/no/nav/syfo/service/DialogmoteInnkallingVarselService.kt

```diff
@@ -0,0 +1,165 @@
+    private val log = LoggerFactory.getLogger(DialogmoteInnkallingVarselService::class.qualifiedName)
+    private val objectMapper = createObjectMapper()
+    fun sendVarselTilNarmesteLeder(varselHendelse: NarmesteLederHendelse) {
+        log.info("[DIALOGMOTE_STATUS_VARSEL_SERVICE]: sender dialogmote hendelse til narmeste leder ${varselHendelse.type}")
+        varselHendelse.data = dataToDialogmoteInnkallingNarmesteLederData(varselHendelse.data)
+        sendVarselTilDineSykmeldte(varselHendelse)
+        sendVarselTilArbeidsgiverNotifikasjon(varselHendelse)
+    }
+
+    fun sendVarselTilArbeidstaker(varselHendelse: ArbeidstakerHendelse) {
+        val url = URL(dialogmoterUrl + BRUKERNOTIFIKASJONER_DIALOGMOTE_SYKMELDT_URL)
+        val text = getArbeidstakerVarselText(varselHendelse.type)
+        senderFacade.sendTilBrukernotifikasjoner(
+            UUID.randomUUID().toString(),
+            varselHendelse.arbeidstakerFnr,
+            text,
+            url,
+            varselHendelse
+        )
+    }
+
+    private fun sendVarselTilArbeidsgiverNotifikasjon(varselHendelse: NarmesteLederHendelse) {
+        val texts = getArbeisgiverTexts(varselHendelse)
+        val sms = texts[SMS_KEY]
+        val emailTitle = texts[EMAIL_TITLE_KEY]
+        val emailBody = texts[EMAIL_BODY_KEY]
+
+        if (!sms.isNullOrBlank() && !emailTitle.isNullOrBlank() && !emailBody.isNullOrBlank()) {
+            val input = ArbeidsgiverNotifikasjonInput(
+                UUID.randomUUID(),
+                varselHendelse.orgnummer,
+                varselHendelse.narmesteLederFnr,
+                varselHendelse.arbeidstakerFnr,
+                ARBEIDSGIVERNOTIFIKASJON_DIALOGMOTE_MERKELAPP,
+                sms,
+                emailTitle,
+                emailBody,
+                LocalDateTime.now().plusWeeks(WEEKS_BEFORE_DELETE)
+            )
+
+            senderFacade.sendTilArbeidsgiverNotifikasjon(
+                varselHendelse,
+                input
+            )
+        }
+    }
+
+    private fun sendVarselTilDineSykmeldte(varselHendelse: NarmesteLederHendelse) {
+        val varselText = getDineSykmeldteVarselText(varselHendelse.type)
+        if (varselText.isNotBlank()) {
+            val dineSykmeldteVarsel = DineSykmeldteVarsel(
+                ansattFnr = varselHendelse.arbeidstakerFnr,
+                orgnr = varselHendelse.orgnummer,
+                oppgavetype = varselHendelse.type.toDineSykmeldteHendelseType().toString(),
+                lenke = null,
+                tekst = varselText,
+                utlopstidspunkt = OffsetDateTime.now().plusWeeks(WEEKS_BEFORE_DELETE)
+            )
+            senderFacade.sendTilDineSykmeldte(varselHendelse, dineSykmeldteVarsel)
+        }
+    }
+
+    private fun getArbeidstakerVarselText(hendelseType: HendelseType): String {
+        return when (hendelseType) {
+            SM_DIALOGMOTE_INNKALT -> BRUKERNOTIFIKASJONER_DIALOGMOTE_INNKALT_TEKST
```

Comment:
Trenger ikke gjøre det nå, men kunne man heller hatt disse tingene i selve enumene? Så slipper vi å ha eksterne funksjoner for å hente tekst osv.